src/SourceGenerators/SourceGeneratorHelpers/Helpers/GeneratorExecutionContextExtensions.cs

```diff
@@ -67,5 +68,17 @@ public string GetMetadataValue(string name)
 
 			return string.IsNullOrEmpty(metadataValue) ? "" : metadataValue!;
 		}
+
+		public static IEqualityComparer<MSBuildItem> IdentityComparer
+			=> _identityComparer ??= new MSBuildItemIdentityComparer();
+
+		private class MSBuildItemIdentityComparer : IEqualityComparer<MSBuildItem>
+		{
+			public bool Equals(MSBuildItem x, MSBuildItem y)
+				=> string.Equals(x.Identity, y.Identity, System.StringComparison.OrdinalIgnoreCase);
```

Comment:
Not sure if we should use StringComparison based on file system's case sensitivity?

Reply:
I wondered, the problem is that it's not easy to determine if the file system should be case sensitive, as it depends on the file location. In our case, having case sensitivity differences is likely to cause other bugs.